test/hotspot/jtreg/vmTestbase/nsk/share/gc/gp/GarbageUtils.java

```diff
@@ -86,6 +87,22 @@ public boolean accept(String errorMessage) {
         private GarbageUtils() {
         }
 
+        public static void engageGC(ExecutionController stresser, long testMemory) {
+            // Number of Young GC iterations before triggering full GC.
+            final long YOUNG_GC_ITERATIONS = 100;
+            final long memChunk = testMemory / YOUNG_GC_ITERATIONS;
+            int iteration = 0;
+
+            while (stresser.continueExecution()) {
```

Comment:
stresser runs for the duration of the test, I believe. Loop completes when test duration is complete.

Reply:
Loop won't run for just once. It will run for the duration of the test. If the test is configured to run, say, 2 mins, stresser.continueExecution() will be false after 2 mins.

Reply:
exactly, however, the whole 'engageGC' method will be executed only once and always work in this loop.
So the combination of lock/engageGC()/unlock is executed only once and the test spends all it's time in 'engageGC' which is not the stress for lock/unlock and moreover could cause OOME if lock/unlock really locks GC.

Reply:
so this is the algorithm you are getting at?
while( test.duration ) {
  lock
  engageGC (100 youngGC and 1 fullGC)
  unlock
}

Reply:
Yes. It is what test should do.